samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                    Unsafe.CopyBlock(pVertexDataBegin, triangleVertices, vertexBufferSize);
+                    _vertexBuffer->Unmap(0, null);
+
+                    // Initialize the vertex buffer view.
+                    _vertexBufferView.BufferLocation = _vertexBuffer->GetGPUVirtualAddress();
+                    _vertexBufferView.StrideInBytes = (uint)sizeof(Vertex);
+                    _vertexBufferView.SizeInBytes = vertexBufferSize;
+                }
+
+                // Create the constant buffer.
+                {
+                    fixed (ID3D12Resource** pConstantBuffer = &_constantBuffer)
+                    {
+                        var heapProperties = new D3D12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_UPLOAD);
+                        var bufferDesc = D3D12_RESOURCE_DESC.Buffer(1024 * 64);
+                        iid = IID_ID3D12Resource;
+                        ThrowIfFailed(nameof(ID3D12Device.CreateCommittedResource), _device->CreateCommittedResource(
+                            &heapProperties,
+                            D3D12_HEAP_FLAG_NONE,
+                            &bufferDesc,
+                            D3D12_RESOURCE_STATE_GENERIC_READ,
+                            pOptimizedClearValue: null,
+                            &iid,
+                            (void**)pConstantBuffer));
+                    }
+
+                    // Describe and create a constant buffer view.
+                    var cbvDesc = new D3D12_CONSTANT_BUFFER_VIEW_DESC {
+                        BufferLocation = _constantBuffer->GetGPUVirtualAddress(),
+                        SizeInBytes = (uint)((sizeof(SceneConstantBuffer) + 255) & ~255)    // CB size is required to be 256-byte aligned.
+                    };
+                    _device->CreateConstantBufferView(&cbvDesc, _cbvHeap->GetCPUDescriptorHandleForHeapStart());
+
+                    // Map and initialize the constant buffer. We don't unmap this until the
+                    // app closes. Keeping things mapped for the lifetime of the resource is okay.
+                    var readRange = new D3D12_RANGE(); // We do not intend to read from this resource on the CPU.
+                    fixed (byte** ppCbvDataBegin = &_pCbvDataBegin)
+                    {
+                        fixed (SceneConstantBuffer* pConstantBufferData = &_constantBufferData)
+                        {
+                            ThrowIfFailed(nameof(ID3D12Resource._Map), _constantBuffer->Map(Subresource: 0, &readRange, (void**)ppCbvDataBegin));
+                            Unsafe.CopyBlock((void*)*ppCbvDataBegin, (void*)pConstantBufferData, (uint)sizeof(SceneConstantBuffer)); // C++: memcpy(_pCbvDataBegin, &_constantBufferData, sizeof(_constantBufferData));
+                            _constantBuffer->Unmap(0, null);
+                        }
+                    }
```

Comment:
The native code never unmaps and as with the previous, we can just use `Unsafe.As` to avoid pinning:
```suggestion
            ThrowIfFailed(nameof(ID3D12Resource.Map), _constantBuffer->Map(Subresource: 0, &readRange, (void**)ppCbvDataBegin));
            Unsafe.CopyBlock(ref _pCbvDataBegin[0], ref Unsafe.As<SceneConstantBuffer, byte>(ref _constantBufferData), (uint)sizeof(SceneConstantBuffer));
```